src/plugin.ts

```diff
@@ -55,6 +55,8 @@ interface UserOIDCAuthState {
   // The `sub` and `aud` claims in the ID token of the last-received
   // TokenSet, if any.
   lastIdTokenClaims?: Pick<IdTokenClaims, 'aud' | 'sub'>;
+  // No ID token was received previously
```

Comment:
This is a bit nit-picky, but just based on this description, this would read a bit like it would start out in a `true` state, because when no token set has been received at all, then also no ID token would have been received previously anyway 

```suggestion
  // No ID token was received in the previous token set
```

If we do make `lastIdTokenClaims` and `noIdTokenReceived` separate properties, we should probably also add a comment about `noIdTokenReceived: true` being incompatible with `lastIdTokenClaims:` being set (the main advantage of giving `lastIdTokenClaims` a special value instead of using a separate property would be that invalid states would not even be representable).

Reply:
yes, that's true, there's a risk of inconsistent state :/ I tried to avoid this with changing the type into something like `type UserOIDCAuthState = UserOIDCAuthStateWithIdToken | UserOIDCAuthStateWithoutIdToken;` (both extending a base type that had most of the properties), but this didn't really seem to prevent the wrong code. I can add the comment.

Reply:
Well, if you want to do that, and if you feel strongly that `lastIdTokenClaims` should not be extended to cover this case (I still think that would be fine as an option, and would keep the code a bit simpler overall as well), you could use something like 

```js
lastIdTokenInfo?: {
  receivedIdToken: true;
  claims: Pick<IdTokenClaims, 'aud' | 'sub'>;
} | {
  receivedIdToken: false;
  claims?: never;
}
```

(this would require a bump of `oidcPluginStateVersion`, though)

Reply:
I considered something like: `lastIdTokenClaims?: Pick<IdTokenClaims, 'aud' | 'sub'> | 'no-id-token';`, but that would also require changes in the serializing, as it's expecting an object. 
Perhaps `lastIdTokenClaims?: Pick<IdTokenClaims, 'aud' | 'sub'> | { no_id_token: true };`? - looks a bit weird, but it's consistently an object, doesn't mix the two versions, and also seems more clear than null values on aud and sub

Reply:
> I considered something like: `lastIdTokenClaims?: Pick<IdTokenClaims, 'aud' | 'sub'> | 'no-id-token';`, but that would also require changes in the serializing, as it's expecting an object. 

Fwiw, I think this is also totally fine – we could extend the deserialization code here fairly easily to account for this as well. (In that case we could bump `oidcPluginStateVersion` to `1` but still keep the ability to deserialize `oidcPluginStateVersion === 0`)

> Perhaps `lastIdTokenClaims?: Pick<IdTokenClaims, 'aud' | 'sub'> | { noIdToken: true };`? - looks a bit weird, but it's consistently an object, doesn't mix the two versions, and also seems more clear than null values on aud and sub (which I trust your words would never happen, but solely from the code I would've not known that)

Also, yes, totally fine, and not all that odd tbh – and I agree with your concern, it wouldn't be obvious from the code itself that that situation could never happen (and even worse, `undefined` rather than `null` _could_ in theory be a value we receive from the IdP).

Reply:
New learning for me:
TS didn't like the original definition, but it works as expected in this format ->
```
type LastIdTokenClaims =
  | (Pick<IdTokenClaims, 'aud' | 'sub'> & { noIdToken?: never })
  | { noIdToken: true };
```

Reply:
New learning for me - TS didn't like the original definition, but this works as expected:
```
type LastIdTokenClaims =
  | (Pick<IdTokenClaims, 'aud' | 'sub'> & { noIdToken?: never })
  | { noIdToken: true };
```